code/modules/mob/living/simple_animal/hostile/necromorph.dm

```diff
@@ -0,0 +1,230 @@
+/*Necromorphs
+	4 types
+		Slasher, melee based, simple mobs
+		Leaper, melee based, high mobility, latch onto foes, hide in vents
+		Puker, semi-ranged based, vomits a highly corrosive cone of acid forwards towards its victims
+		Exploder, melee based, steady shuffle towards a target before exploding. Explodes on death
+*/
+/mob/living/simple_animal/hostile/necromorph
+	name = "necromorph"
+	desc = "A twisted husk of what was once human, repurposed to kill."
+	speak_emote = list("roars")
+	icon = 'icons/mob/monster_big.dmi'
+	icon_state = "nmorph_standard"
+	icon_living = "nmorph_standard"
+	icon_dead = "nmorph_dead"
+	health = 80
+	maxHealth = 80
+	melee_damage_lower = 25
+	melee_damage_upper = 50
+	attacktext = "slashes"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	faction = "marker"
+	speed = 4
+	size = SIZE_BIG
+	move_to_delay = 4
+	canRegenerate = 1
+	minRegenTime = 300
+	maxRegenTime = 1200
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+
+/mob/living/simple_animal/hostile/necromorph/leaper
+	desc = "A twisted husk of what was once human. Sporting razor-sharp fangs, along with a long scythe-tipped tail."
+	icon_state = "nmorph_leaper"
+	icon_living = "nmorph_leaper"
+	icon_dead = "nmorph_leaper_dead"
+	speed = 1
+	health = 45
+	maxHealth = 45
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	attacktext = "slashes"
+	attack_sound = 'sound/weapons/slashmiss.ogg'
+
+	minRegenTime = 240
+	maxRegenTime = 600
+
+	ranged = 1
+	ranged_cooldown_cap = 8
+	ranged_message = "leaps"
+
+/mob/living/simple_animal/hostile/necromorph/leaper/Shoot(var/atom/target, var/atom/start, var/mob/user, var/bullet = 0)
+	if(locked_to)
+		return 0
+
+	src.throw_at(get_turf(target),7,1)
+	return 1
+
+/mob/living/simple_animal/hostile/necromorph/leaper/Bump(atom/A)
+	if(throwing && isliving(A) && CanAttack(A))
+		attach(A)
+	..()
+
+/mob/living/simple_animal/hostile/necromorph/leaper/Life()
+
+	update_climb()
+	if(!isUnconscious())
+		if(stance == HOSTILE_STANCE_IDLE && !client)
+			var/list/can_see = view(get_turf(src), vision_range/1.5) //Nothing too close for comfort
```

Comment:
Yeah 2 should be fine.  I'm not sure if it makes a difference.  Still, the range they detect you and leave the vent is 3-4 tiles which is way way more than enough for the player to say nope